src/common/state/table.c

```diff
@@ -101,8 +96,11 @@ int64_t table_timeout_handler(event_loop *loop,
     return EVENT_LOOP_TIMER_DONE;
   }
 
-  /* Decrement retry count and try again. */
-  callback_data->retry.num_retries--;
+  /* Decrement retry count and try again. We use -1 to indicate infinite
+   * retries. */
+  if (callback_data->retry.num_retries != -1) {
```

Comment:
Actually, what if we hold off on this for now? I'd like to give it a little more thought.